### PR TITLE
Nlu ouput path results

### PR DESCRIPTION
**Pull request by ismaeelBM**
**Proposed changes**:
- Fix for Issue #4144

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black`

### CODE REVIEW

**Comment by msamogh** [comment]
Thanks for the PR, we'll give it a review as soon as possible.

**Comment by ismaeelBM** [comment]
@tabergma Oh sorry I completely missed those. So after looking through the arguments file, I had a question. I realize that the nlu also have a reports folder arguments. Would you like me to code it so that the argument for reports folder is treated as a child directory to the output folder for the NLU? Or do you want the report and output to be treated as two completely separate directory?

**Comment by tabergma** [comment]
Good catch 👍Let's use a separate folder for the output. So, the report folder should become a child directory of the output directory.

**Comment by ismaeelBM** [comment]
@tabergma I have made the changes. hopefully this will be able to address the cases. Let me know if any further changes are needed.

File: rasa/cli/test.py [hunk 1]
```diff
@@ -100,6 +100,10 @@ def test_nlu(args: argparse.Namespace) -> None:
 
     nlu_data = cli_utils.get_validated_path(args.nlu, "nlu", DEFAULT_DATA_PATH)
     nlu_data = data.get_nlu_directory(nlu_data)
+    output = args.out or DEFAULT_RESULTS_PATH
+
+    if not os.path.exists(output):
```
Comment (tabergma) [inline rasa/cli/test.py]: You can use 
```
    io_utils.create_directory(output)
```

File: rasa/cli/test.py [hunk 2]
```diff
@@ -131,11 +135,15 @@ def test_nlu(args: argparse.Namespace) -> None:
                 )
                 continue
 
-        output = args.report or DEFAULT_NLU_RESULTS_PATH
+        report_output = args.report or DEFAULT_NLU_RESULTS_PATH
+
+        if output:
```
Comment (tabergma) [inline rasa/cli/test.py]: `output` will always be set, no need for `if`

File: rasa/test.py [hunk 3]
```diff
@@ -109,7 +109,12 @@ def test_core(
     )
 
 
-def test_nlu(model: Optional[Text], nlu_data: Optional[Text], kwargs: Optional[Dict]):
+def test_nlu(
+    model: Optional[Text],
+    nlu_data: Optional[Text],
+    kwargs: Optional[Dict],
+    output: Text = DEFAULT_RESULTS_PATH,
```
Comment (tabergma) [inline rasa/test.py]: Minor: I would like to keep `kwargs` as last argument. Can you change the order?

File: rasa/test.py [hunk 4]
```diff
@@ -122,11 +127,14 @@ def test_nlu(model: Optional[Text], nlu_data: Optional[Text], kwargs: Optional[D
         )
         return
 
+    if output:
```
Comment (tabergma) [inline rasa/test.py]: `output` is always set, no need for `if`.

File: rasa/cli/test.py [hunk 5]
```diff
@@ -131,11 +135,15 @@ def test_nlu(args: argparse.Namespace) -> None:
...
+            report_output = os.path.join(output, report_output)
```
Comment (tabergma) [inline rasa/cli/test.py]: I guess, we don't need that. `--report` is used for `Output path to save the intent/entity metrics report.`. So, if we do compare testing, I would store the results just in the `--out` folder. No need to "misuse" the report folder here anymore.

File: rasa/cli/test.py (hunk 1 above)
Comment (ismaeelBM) [reply rasa/cli/test.py]: The same command was already used in the test_core function of that file. Would you like me to update that too?

Comment (tabergma) [reply rasa/cli/test.py]: Sure. Thanks!

**Comment by ismaeelBM** [comment]
@tabergma I am all done with the changes! Let me know if there is anymore! Also I think I found a bug when in the argument for config you use a folder as input instead of file names. At first I thought it was due to some bug in my code then I realized it was also present in the master. I will post that issue by tonight.

**Pull request merged**